src/main.cpp

```diff
@@ -147,51 +147,77 @@ int main(int argc, char** argv)
     auto applyCliArgs = [&](Settings* settings) {
         if (parser.isSet(sysroot)) {
             settings->setSysroot(parser.value(sysroot));
+            settings->setLastUsedEnvironment({});
         }
         if (parser.isSet(kallsyms)) {
             settings->setKallsyms(parser.value(kallsyms));
+            settings->setLastUsedEnvironment({});
         }
         if (parser.isSet(debugPaths)) {
             settings->setDebugPaths(parser.value(debugPaths));
+            settings->setLastUsedEnvironment({});
         }
         if (parser.isSet(extraLibPaths)) {
             settings->setExtraLibPaths(parser.value(extraLibPaths));
+            settings->setLastUsedEnvironment({});
         }
         if (parser.isSet(appPath)) {
             settings->setAppPath(parser.value(appPath));
+            settings->setLastUsedEnvironment({});
         }
         if (parser.isSet(arch)) {
             settings->setArch(parser.value(arch));
+            settings->setLastUsedEnvironment({});
         }
     };
 
     const auto settings = Settings::instance();
+    settings->loadFromFile();
     applyCliArgs(settings);
-    for (const auto& file : parser.positionalArguments()) {
-        auto window = new MainWindow;
+
+    auto files = parser.positionalArguments();
+    auto arguments = app.arguments();
+    arguments.pop_front(); // pop executable
+
+    for (int i = 0, s = files.size(); i < s; i++) {
+        arguments.removeLast();
+    }
+
+    while (files.size() > 1) {
+        // spawn new instances if we have more than one file argument
+        QProcess process;
+        process.setProgram(app.applicationFilePath());
+        const auto file = files.front();
+        files.pop_front();
+        auto args = arguments;
+        args.append(file);
+        process.setArguments(args);
+        if (!process.startDetached()) {
+            qWarning() << "failed to start new hotspot instance";
```

Comment:
qWarning() << "failed to start new hotspot instance for file" << file << process.errorString();